pkg/client/cache/reflector_test.go

```diff
@@ -360,3 +360,43 @@ func TestReflector_ListAndWatchWithErrors(t *testing.T) {
 		r.ListAndWatch(util.NeverStop)
 	}
 }
+
+func TestReflectorResync(t *testing.T) {
```

Comment:
Is it possible to write a test that verifies that the connection actually gets reused? I worry that someone who breaks this test is going to be able to fix it while still not managing to get the connection reused.